docs/en/advanced_tutorials/basedataset.md

```diff
@@ -1,3 +1,503 @@
+
+```python
+pipeline = [
+    LoadImage(),
+    ParseImage(),
+]
+
+toy_dataset = ToyDataset(
+    data_root='data/',
+    data_prefix=dict(img_path='train/'),
+    ann_file='annotations/train.json',
+    pipeline=pipeline,
+    # Pass the lazy_init variable in here
+    lazy_init=True)
+```
+
+When `lazy_init=True`, the initialization of ToyDataset's only performs steps 1, 2, and 3 of the BaseDataset initialization process. At this time, `toy_dataset` was not fully initialized, since `toy_dataset` will not read and parse the annotation file. The `toy_dataset` only set the meta information of the dataset (`metainfo`).
+
+Naturally, if you need to access specific data information later, you can manually call the `toy_dataset.full_init()` interface to perform the complete initialization process, during which the data annotation file will be read and parsed. Calling the `get_data_info (independence idx)`, `__len__ ()`, `__getitem__ (independence idx)`, ` get_subset_ (indices)` and `get_subset(indices)` interface will also automatically call the `full_init()` interface to perform the full initialization process (only on the first call, later calls will not call the `full_init()` interface repeatedly):
+
+```python
+# Full initialization
+toy_dataset.full_init()
+
+# After initialization, you can now get the data info
+len(toy_dataset)
+# 2
+toy_dataset[0]
+# {
+#     'img_path': "data/train/xxx/xxx_0.jpg",
+#     'img_label': 0,
+#     'img': a ndarray with shape (H, W, 3), which denotes the value the image,
+#     'img_shape': (H, W, 3) ,
+#     ...
+# }
+```
+
+**Notice:**
+
+Performing full initialization by calling the `__getitem__()` interface directly carries some risks: If a dataset class is not fully initialized by setting `lazy_init=True` firstly, then it is directly sent to the dataloader. In the subsequent data reading process, different dataloader workers will read and parse the annotation file at the same time. Although this may work normally, it consumes a lot of time and memory. **Therefore, it is recommended to manually call the `full_init()` interface to perform the full initialization process before you need to access specific data.**
+
+The above is not fully initialized by setting `lazy_init=True`, and then complete initialization according to the demand, called lazy init.
+
+### Save memory
+
+In the specific process of reading data, the dataloader will usually prefetch data from multiple dataloader workers, and multiple workers have complete dataset class backup, so there will be multiple copies of the same `data_list` in the memory. In order to save this part of memory consumption, The BaseDataset can serialize `data_list` into memory in advance, so that multiple workers can share the same copy of `data_list`, so as to save memory.
```

Comment:
```suggestion
In the specific process of reading data, the dataloader will usually prefetch data from multiple dataloader workers, and multiple workers have complete dataset object backup, so there will be multiple copies of the same `data_list` in the memory. In order to save this part of memory consumption, The `BaseDataset` can serialize `data_list` into memory in advance, so that multiple workers can share the same copy of `data_list`, so as to save memory.
```